components/data-bridge/org.wso2.carbon.databridge.receiver.thrift/src/main/java/org/wso2/carbon/databridge/receiver/thrift/internal/ThriftServerStartupImpl.java

```diff
@@ -27,11 +28,18 @@
 /**
  * Thrift Server Startup Implementation.
  */
-public class ThriftServerStartupImpl implements ThriftServerStartup {
+public class ThriftServerStartupImpl implements ServerEventListener {
 
     private static final Logger log = Logger.getLogger(ThriftServerStartupImpl.class);
+    private static final String DISABLE_RECEIVER = "disable.receiver";
 
-    public void completingServerStartup() {
+    @Override
+    public void start() {
+        String disableReceiver = System.getProperty(DISABLE_RECEIVER);
+        if (disableReceiver != null && Boolean.parseBoolean(disableReceiver)) {
```

Comment:
`Boolean.parseBoolean()` has a null check inside, so we do not need to do a null check I guess